### ISSUE TITLE
Support 30-day trailing total event count

### ISSUE THREAD

**Issue by jonyTF** [PostHog/posthog#20430]
## Is your feature request related to a problem?

N/A

## Describe the solution you'd like

I want to be able to display the 30-day trailing total event count on a graph. For example, I have a "Group Created" event, and I want to see how many of these events have happened in the last 30 days, and I want that "last 30 days" metric to update every day. This is currently possible for users (i.e. "monthly active users"), but not for events.

## Describe alternatives you've considered

Currently I have a table set up that calculates total event count for the last 30 days, and I've been manually checking it every so often and creating a graph of the value over time in Google Sheets 💀

Would love to have PostHog automatically do it for me :)

## Additional context

N/A

**Comment by mariusandra** [comment]
Hey, is this "smoothing" what you're looking for?

<img width="710" alt="image" src="https://github.com/PostHog/posthog/assets/53387/5272979c-0a59-416a-ae61-f2e21ff744d6">

**Comment by jonyTF** [comment]
No, "smoothing" does not seem to do what I want... Basically I want to view the total number of times an event has fired in the last 30 days, and have that metric graphed every day.

**Comment by jonyTF** [comment]
Although seeing the average is nice, I would like to see the raw total event count. Multiplying by 28 does give a general estimate, but is still using an average and thus not the actual amount of events in the last 28 days.

**Comment by mariusandra** [comment]
Mathematically speaking, an average is the total divided by 28, so multiplying it by 28 should get you the same number, save for a potential small floating point rounding error. 🤔

**Comment by jonyTF** [comment]
Hmmmm I seem to be getting two different numbers when I do the smoothing thing and multiply by 28 and when I just get the total count in the last 28 days...

![image](https://github.com/PostHog/posthog/assets/16812809/8a5eb473-d2d2-417c-88c1-a30b86b60184)

![image](https://github.com/PostHog/posthog/assets/16812809/91961682-bbe3-45bc-9a3a-8aa6e8169ba8)

**Comment by mariusandra** [comment]
Hmm... something's a bit off indeed then 🤔. We could dig a bit deeper into the numbers ourselves to see what's what, but for that can you send us a support request via the app, linking to this issue?

There is however another way you can get this same data, if you're familiar with SQL. If you enable the "Business Intelligence" feature preview, then the "SQL insight" tab will let you make a chart out of any data you can query with SQL.

The following query gave me a table of rolling 30 day averages for the last 30 days:

```sql
select a.2 as date, a.1 as count from (
    select arrayJoin(arrayZip(counts, times)) as a from (
        select groupArrayMovingAvg(30)(count) as counts, groupArray(timestamp) as times from (
            select toDate(timestamp) as timestamp, count() as count
            from events
            where timestamp > now() - interval 60 day and timestamp <= now()
            group by timestamp
            order by timestamp asc
            limit 100
        )
    )
) limit 30 offset 30
```

**Comment by jonyTF** [comment]
ooooh thank you, this feature to create my own custom charts is really helpful!

**Comment by slshults** [comment]
+ https://posthoghelp.zendesk.com/agent/tickets/43503 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/45027)
